Fix test execution in torchtext

Nayef211 opened the pull request:
## Description
- Introduces `test/torchtext_unittest` directory and moves all the test scripts and assets there (this is in line with how torchaudio organizes it's unit tests)
- changes `python setup.py develop` to `python setup.py install` in CI job
- run test with (`cd test && pytest torchtext_unittest`) so that repository root is not in Python module search path.
- Change relative import (from `..common import ...`) in test module to absolute import (`from torchtext_unittest.common import ...`)
- Creates a workflow to run integration tests seperately from unittests

## Additional context
- Followup to https://github.com/pytorch/text/pull/1886 
- Stale PR that attempted to do this in the past https://github.com/pytorch/text/pull/1223

## Followups

- [ ] Investigate `test_vocab_from_raw_text_file` failure on linux platforms (https://github.com/pytorch/text/issues/1900)
- [ ] Keeping asset dir and make integration test not access internals of `torchtext_unittest` 

Closes #1875

Nayef211 commented:
> To confirm: Failures related to `test_vocab_from_raw_text_file` will be resolved in a followup PR or in this PR?
The plan is to investigate and resolve them in a followup PR. There aren't any significant changes in this PR that should cause that test to fail.

joecummings reviewed (approved):
lgtm

mthrok commented on tools/setup_helpers/extension.py:
```diff
@@ -21,6 +21,10 @@
 
 
+def _get_cxx11_abi():
+    return "-D_GLIBCXX_USE_CXX11_ABI=" + str(int(torch._C._GLIBCXX_USE_CXX11_ABI))
```
nit: Perhaps we can use `torch.compiled_with_cxx11_abi` instead of `torch._C._GLIBCXX_USE_CXX11_ABI`.

https://github.com/pytorch/pytorch/search?l=Python&q=_GLIBCXX_USE_CXX11_ABI

Nayef211 replied:
Done

mthrok commented on tools/setup_helpers/extension.py:
```diff
@@ -72,6 +76,7 @@ def build_extension(self, ext):
             "-DBUILD_SHARED_LIBS=OFF",
             "-DCMAKE_POLICY_DEFAULT_CMP0063=NEW",
             "-DSPM_ENABLE_SHARED=OFF",
+            f"-DCMAKE_CXX_FLAGS={_get_cxx11_abi()}",
```
I am not sure if `-DCMAKE_CXX_FLAGS` appends the provided value to default values or overwrites them. Can you do the research to make sure it's file, or perhaps use custom option name to avoid potential conflict?

Nayef211 replied:
Added a cache variable to store this value. Lmk if this looks okay to you.

mthrok reviewed (commented):
Overall it looks good.

* [test/.gitignore → test/torchtext_unittest/.gitignore](https://github.com/pytorch/text/pull/1889/files#diff-383dfb4aa44afcd0ee5b6451525e145997d6ae6691281989f50762e1541ee57f): This file is empty. Do we need it?

* Potential follow-up: keeping `asset` dir and make integration test not access internals of `torchtext_unittest` would make it cleaner.

mthrok commented:
> > To confirm: Failures related to `test_vocab_from_raw_text_file` will be resolved in a followup PR or in this PR?
> 
> The plan is to investigate and resolve them in a followup PR. There aren't any significant changes in this PR that should cause that test to fail.

Was it failing before?

Nayef211 commented:
> > > To confirm: Failures related to `test_vocab_from_raw_text_file` will be resolved in a followup PR or in this PR?
> > 
> > 
> > The plan is to investigate and resolve them in a followup PR. There aren't any significant changes in this PR that should cause that test to fail.
> 
> Was it failing before?

Yes it was failing sporadically on certain python versions on Linux machines. Disabled them for the time being in https://github.com/pytorch/text/pull/1901

Nayef211 commented:
> * [test/.gitignore → test/torchtext_unittest/.gitignore](https://github.com/pytorch/text/pull/1889/files#diff-383dfb4aa44afcd0ee5b6451525e145997d6ae6691281989f50762e1541ee57f): This file is empty. Do we need it?
Deleted the file 
> * Potential follow-up: keeping `asset` dir and make integration test not access internals of `torchtext_unittest` would make it cleaner.
Added it as a followup item in PR description
